meson/arch/amd64/meson.build

```diff
@@ -0,0 +1,46 @@
+# TODO: Red Zone is only a problem if we allow interrupts to land on an active
+#       stack. Using separate stack for interrupts and ensuring that we never
+#       enable interrupts on an interrupt stack would allow us to get rid of
+#       the argument.
+
+arch_uspace_c_args = [
+	'-D__LE__',
+	'-fno-omit-frame-pointer',
+]
+
+arch_kernel_c_args = arch_uspace_c_args + [
+	'-mno-sse',
+	'-mno-sse2',
+	'-mcmodel=' + MEMORY_MODEL,
+	'-mno-red-zone',
+	'-fno-unwind-tables',
+]
+
+if PROCESSOR == 'opteron'
+	arch_kernel_c_args += '-march=opteron'
+endif
+
+
+# TODO: Enable --gc-sections
```

Comment:
So what prevents it?

Reply:
I think the issue is that the linker doesn't know that multiboot entry point is referenced, a therefore eliminates all code. It shouldn't be too hard to fix.